tests/test_canonicaljson.py

```diff
@@ -167,3 +152,40 @@ def test_set_json(self) -> None:
             from canonicaljson import json  # type: ignore[attr-defined]
 
             set_json_library(json)
+
+    def test_encode_unknown_class_raises(self) -> None:
+        class C:
+            pass
+
+        with self.assertRaises(Exception):
+            encode_canonical_json(C())
+
+    def test_preserialisation_callback(self) -> None:
+        class C:
+            pass
+
+        register_preserialisation_callback(C, lambda c: "I am a C instance")
```

Comment:
Note that this call (and below) naughtily alters module global state. Probably wants a warning comment or something.